system/src/system.cpp

```diff
@@ -66,6 +69,52 @@ int lua_threadcount(lua_State* L)
     return 1;
 }
 
+int lua_freememory(lua_State* L)
+{
+    lua_pushinteger(L, uv_get_free_memory()); // FIXME: this is a 64 bit int, deal with it?
+
+    return 1;
+}
+
+int lua_totalmemory(lua_State* L)
+{
+    lua_pushinteger(L, uv_get_total_memory()); // FIXME: this is a 64 bit int, deal with it?
+
+    return 1;
+}
+
+int lua_hostname(lua_State* L)
+{
+    size_t sz = UV_MAXHOSTNAMESIZE;
+    char* hostname = (char*) malloc(sz);
```

Comment:
Use std::string, don't make a C-style buffer. `process.cwd` is an example you could look to. You could also just copy the `cwd` implementation and place it here, they're extremely similar.